hibernate-core/src/main/java/org/hibernate/query/programmatic/QuerySpecification.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.query.programmatic;
+
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Incubating;
+import org.hibernate.query.restriction.Restriction;
+
+/**
+ * Commonality for all query specifications.
+ *
+ * @apiNote Query specifications only support a {@linkplain #getQueryRoot() single root}.
+ *
+ * @author Steve Ebersole
+ */
+@Incubating
+public interface QuerySpecification<T> {
+	/**
+	 * Get the root of the query.
+	 * E.g. given the HQL {@code "from Book"}, we have a single {@code Root<Book>}.
+	 */
+	Root<T> getQueryRoot();
```

Comment:
```suggestion
	Root<T> getQueryRoot();
	
	CommonAbstractCriteria getQuery();
```

Reply:
Given the initial impl, exposing the AST to the user for possible maipulation was bad.  But considering the way this has evolved, I guess that's not unreasonable.

Reply:
Yeah I mean we *do* already expose a way to do this via `HibernateCriteriaBuilder.createQuery()`, but this is a more convenient API, and *much* more easily discoverable for the user.

Reply:
I wonder though if this is not a place we want to expose `JpaCriteriaBase`.  Our layer has better support for common selection/mutation definition - here specifically `JpaCriteriaQuery` and `JpaManipulationCriteria`

Reply:
And given that, possibly `s/Root/JpaRoot` as well.

Reply:
My knee-jerk reaction would be to stick with the `Root` for now. They can easily cast if they need to.

Reply:
> Yeah I mean we _do_ already expose a way to do this via `HibernateCriteriaBuilder.createQuery()`, but this is a more convenient API, and _much_ more easily discoverable for the user.

Expose what exactly?  `HibernateCriteriaBuilder.createQuery()` is simply creating a new criteria.  This is a different beast.  Like I said, given how the code/design is now it is not terrible to expose the tree.

Reply:
Expose an operation for going from HQL to `CriteriaQuery`. Just saying that it's convenient to also have that operation exposed here.